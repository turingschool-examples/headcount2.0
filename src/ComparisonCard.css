.comparison-card {
  background-color: #9b9b9b;
  border: 8px solid black;
  border-radius: 20px;
  display: flex;
  flex-direction: column;
  height: 270px;
  justify-content: space-evenly;
  margin: 10px;
  opacity: 0.9;
  padding-bottom: 10px;
  text-align: center;
  width: 280px;
}

.comparison-card h3 {
  font-size: 20px;
  font-weight: bold;
  line-height: 2.8;
  margin-bottom: 0;
  margin-top: 0;
  padding: 0;
  position: relative;
}

.comparison-card p {
  background-color: lightgray;
  border: 2px solid black;
  border-radius: 20px;
  font-size: 16px;
  font-weight: bold;
  margin: auto;
  padding: 4px;
  width: 60px;
}

.avg-bottom {
  margin-left: 40px;
}

.avg-top {
  margin-right: 40px;
}

.pencil-icon {
  height: 42px;
  margin-right: 26px;
  position: relative;
  top: 13px;
  transform: rotate(45deg);
}

.rotate {
  margin-left: 26px;
  transform: rotate(225deg);
}

.apple-placeholder {
  background-color: #9b9b9b;
  border: 8px solid black;
  border-radius: 20px;
  height: 280px;
  margin: 10px;
  opacity: 0.9;
  width: 280px;
}

.apple-pic {
  display: flex;
  height: 150px;
  margin: 55px auto 0 auto;
}

@media screen and (max-width: 905px) {
  .pencil-icon {
    transform: rotate(135deg);
  }

  .rotate {
    transform: rotate(315deg);
  }
}
