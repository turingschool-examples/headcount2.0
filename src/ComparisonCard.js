import React from "react";

export const ComparisonCard = ({ comparisonData }) => {
  const comparisonKeys = Object.keys(comparisonData);
  const districtKeyOne = comparisonKeys[0];
  const districtKeyTwo = comparisonKeys[1];
  return (
    <div>
      <p>
        {comparisonKeys[0]} :<span> {comparisonData[districtKeyOne]}</span>
      </p>
      <container>
        ----
        {comparisonData[comparisonKeys[2]]}
        ----
      </container>
      <p>
        {comparisonKeys[1]} :<span> {comparisonData[districtKeyTwo]}</span>
      </p>
    </div>
  );
};
