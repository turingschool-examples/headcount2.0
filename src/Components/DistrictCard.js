import React from "react";
import "../CSS/DistrictCard.css";
import PropTypes from "prop-types";

export const DistrictCard = ({
  location,
  stats,
  clicked,
  selectCard,
  counter
}) => {
  return (
    <div
      onClick={() => selectCard(location)}
      className={clicked ? "district-card-clicked" + counter : "district-card"}
    >
      <h1 className="district-location">{location}</h1>
      <header className={clicked ? "year-header-show" : "year-header-hide"}>
        Year
      </header>
      <header className={clicked ? "avg-header-show" : "avg-header-hide"}>
        Avg
      </header>
      {Object.keys(stats).map((year, index) => {
        const avg = stats[year];
        return (
          <section
            className={clicked ? "stats-show" : "stats-hide"}
            key={index}
          >
            <p className="year">
              {year}
              ......
              <span className={avg < 0.5 ? "avg-low" : "avg-high"}>{avg}</span>
            </p>
          </section>
        );
      })}
      <p className={clicked ? "show-msg" : "hide-msg"}>Click to Remove</p>
    </div>
  );
};

const { string, shape, number, bool, func } = PropTypes;

DistrictCard.propTypes = {
  location: string,
  stats: shape({ year: number }),
  clicked: bool,
  selectCard: func,
  counter: number
};
