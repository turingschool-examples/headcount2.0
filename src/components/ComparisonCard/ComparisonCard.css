.compare-card h1 {
  font-size: 3em;
  margin: 0;
}

.avg {
  font-size: 1.5em;
  color: #78C9D0 !important;
}

.bottom {
  margin-bottom: 0;
}

.name-target {
  margin: 35px 0 10px 0;
}