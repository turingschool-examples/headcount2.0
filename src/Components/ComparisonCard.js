import React from "react";
import PropTypes from "prop-types";
import "../CSS/ComparisonCard.css";

export const ComparisonCard = ({ comparisonData }) => {
  const comparisonKeys = Object.keys(comparisonData);
  const districtKeyOne = comparisonKeys[0];
  const districtKeyTwo = comparisonKeys[1];
  return (
    <div className="compare-data-card">
      <div className="district-title">
        Aggregate Avg.
        <p
          className={
            comparisonData[districtKeyOne] < comparisonData[districtKeyTwo]
              ? "district-low"
              : "district-high"
          }
        >
          {comparisonData[districtKeyOne]}
        </p>
      </div>
      <p className="comparison-value">{comparisonData[comparisonKeys[2]]}</p>
      <div className="district-title">
        Aggregate Avg.
        <p
          className={
            comparisonData[districtKeyOne] > comparisonData[districtKeyTwo]
              ? "district-low"
              : "district-high"
          }
        >
          {comparisonData[districtKeyTwo]}
        </p>
      </div>
    </div>
  );
};

const { shape, number } = PropTypes;

ComparisonCard.propTypes = {
  comparisonData: shape({ year: number })
};
