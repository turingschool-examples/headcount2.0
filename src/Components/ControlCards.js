import React from "react";
import { ComparedDistrictCard } from "./ComparedDistrictCard";
import { ComparisonCard } from "./ComparisonCard";
import PropTypes from "prop-types";
import "../CSS/ControlCards.css";

export const ControlCards = ({
  comparisonData,
  selectCard,
  selectedDistricts
}) => {
  const controlCards = Object.keys(comparisonData).map((district, index) => {
    if (selectedDistricts.length === 2) {
      return comparisonData[district].hasOwnProperty("location") ? (
        <ComparedDistrictCard
          key={`compared ${index}`}
          selectCard={selectCard}
          location={district}
          stats={comparisonData[district].stats}
          index={index}
        />
      ) : (
        <ComparisonCard key={index} comparisonData={comparisonData[district]} />
      );
    }
  });
  return <div className="control-cards">{controlCards} </div>;
};

const { string, number, bool, shape, objectOf, func, arrayOf } = PropTypes;

ControlCards.propTypes = {
  comparisonData: objectOf(
    shape({
      clicked: bool,
      location: string,
      stats: shape({ year: number })
    })
  ),
  selectedDistricts: arrayOf(
    shape({
      clicked: bool,
      location: string,
      stats: shape({ year: number })
    })
  ),
  selectCard: func
};
