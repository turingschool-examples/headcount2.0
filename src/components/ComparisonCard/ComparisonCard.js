import React from 'react';
import './ComparisonCard.css';
import PropTypes from 'prop-types';

const ComparisonCard = ({comparedObject}) => {
  if (comparedObject) {
    const d1Name = Object.keys(comparedObject)[0];
    const d1Avg = comparedObject[Object.keys(comparedObject)[0]];
    const d2Name = Object.keys(comparedObject)[1];
    const d2Avg = comparedObject[Object.keys(comparedObject)[1]];
    const compareRatio = comparedObject.compared;
    
    return (
      <article className="compare-card">
        <h3>{d1Name}</h3>
        <h3>{d1Avg}</h3>
        <h1>{compareRatio}</h1>
        <h3>{d2Name}</h3>
        <h3>{d2Avg}</h3>
      </article>
    );
  }
  
  return (
    <div></div>
  );
};

ComparisonCard.propTypes = {
  comparedObject: PropTypes.object
};

export default ComparisonCard;