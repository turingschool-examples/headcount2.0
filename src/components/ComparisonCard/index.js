import React from "react";
import applePic from "../../images/apple.png";
import pencilIcon from "../../images/pencil.svg";
import PropTypes from "prop-types";
import "./ComparisonCard.css";

const ComparisonCard = props => {
  const districtNames = Object.keys(props);
  if (districtNames[0] === "props" || districtNames[1] === "props") {
    return (
      <div className="apple-placeholder">
        <img className="apple-pic" alt="apple" src={applePic} />
      </div>
    );
  } else {
    return (
      <div className="comparison-card">
        <h3 className="avg-top">
          <img className="pencil-icon" alt="pencil" src={pencilIcon} />
          AVERAGE
        </h3>
        <p> {props[districtNames[0]]}</p>
        <h3>COMPARED AVERAGE</h3>
        <p> {props.compared}</p>
        <h3 className="avg-bottom">
          AVERAGE
          <img className="pencil-icon rotate" alt="pencil" src={pencilIcon} />
        </h3>
        <p> {props[districtNames[1]]}</p>
      </div>
    );
  }
};

ComparisonCard.defaultProps = {
  props: PropTypes.obj
};

ComparisonCard.propTypes = {
  compared: PropTypes.number
};

export default ComparisonCard;
