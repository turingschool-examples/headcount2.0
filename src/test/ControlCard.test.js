import React from "react";
import { ControlCards } from "../Components/ControlCards";
import { ComparisonCard } from "../Components/ComparisonCard";
import { DistrictCard } from "../Components/DistrictCard";

describe("ControlCards", () => {
  let shallowWrap;
  let mountWrap;
  let mockClick = { "BAYFIELD 10 JT-R": true, "BENNETT 29J": true };
  let mockData = {
    "AGUILAR REORGANIZED 6": {
      location: "AGUILAR REORGANIZED 6",
      stats: {
        2004: 1,
        2005: 1
      },
      cardClick: null
    },
    "AKRON R-1": {
      location: "AKRON R-1",
      stats: {
        2008: 0.844,
        2009: 0.96
      },
      cardClick: null
    },
    comparisonData: {
      "AKRON R-1": 0.619,
      "AGUILAR REORGANIZED 6": 1,
      compared: 0.619
    }
  };

  const mockdDistricts = [
    {
      clicked: true,
      location: "COLORADO",
      stats: {
        2004: 0.24,
        2005: 0.278,
        2006: 0.337,
        2007: 0.395,
        2008: 0.536,
        2009: 0.598,
        2010: 0.64,
        2011: 0.672,
        2012: 0.695,
        2013: 0.703,
        2014: 0.741
      }
    },
    {
      clicked: true,
      location: "ACADEMY 20",
      stats: {
        2004: 0.24,
        2005: 0.278,
        2006: 0.337,
        2007: 0.395,
        2008: 0.536,
        2009: 0.598,
        2010: 0.64,
        2011: 0.672,
        2012: 0.695,
        2013: 0.703,
        2014: 0.741
      }
    }
  ];

  beforeEach(() => {
    shallowWrap = shallow(
      <ControlCards
        comparisonData={mockData}
        cardClick={mockClick}
        selectedDistricts={mockdDistricts}
      />
    );
    mountWrap = mount(
      <ControlCards
        comparisonData={mockData}
        cardClick={mockClick}
        selectedDistricts={mockdDistricts}
      />
    );
  });

  it("should exist", () => {
    expect(shallowWrap).toBeDefined();
    expect(mountWrap).toBeDefined();
  });

  it("should match snapshot with no control card", () => {
    shallowWrap = shallow(
      <ControlCards comparisonData={{}} cardClick={mockClick} />
    );

    expect(shallowWrap).toMatchSnapshot();
  });

  it("should match snapshot", () => {
    expect(shallowWrap).toMatchSnapshot();
  });
});
