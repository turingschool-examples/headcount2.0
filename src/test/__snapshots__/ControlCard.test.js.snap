// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlCards should match snapshot 1`] = `
<div
  className="control-cards"
>
  <Component
    index={0}
    key="compared 0"
    location="AGUILAR REORGANIZED 6"
    stats={
      Object {
        "2004": 1,
        "2005": 1,
      }
    }
  />
  <Component
    index={1}
    key="compared 1"
    location="AKRON R-1"
    stats={
      Object {
        "2008": 0.844,
        "2009": 0.96,
      }
    }
  />
  <Component
    comparisonData={
      Object {
        "AGUILAR REORGANIZED 6": 1,
        "AKRON R-1": 0.619,
        "compared": 0.619,
      }
    }
    key="2"
  />
   
</div>
`;

exports[`ControlCards should match snapshot with no control card 1`] = `
<div
  className="control-cards"
>
   
</div>
`;
